Pipfile

```diff
@@ -14,6 +14,7 @@ paho-mqtt = "*"
 enum34 = "*"
 PyYAML = "<=5.2,>=4.2b1"
 Cerberus = "*"
+"rpi.gpio" = "*"
```

Comment:
Is there a reason you've added this to the main requirements file? It should get installed at run-time, since this software (despite its name) is not specific to the Raspberry Pi. Is this not happening for you?

Reply:
@flyte You are right that it should get installed automatically. As mentioned above, I put it here mainly, because the pi I am running this on doesn't have internet access.

The second reason is that I personally consider pulling requirements into docker containers at runtime is an anti-pattern. These dependencies are not part of the image and cannot be audited by conventional static analysis for docker images. They also require the servers that provide the dependency to be highly available or one would risk that a fresh container cannot start. There are other reasons why in general Docker images should not pull dependencies at runtime. For example they won't play nicely in Kubernetes, but I don't think someone wants to create a HA Kubernetes cluster to run mqtt-gpio 😉.

As a compromise [you suggested already](https://github.com/flyte/pi-mqtt-gpio/pull/112#issuecomment-636496737) that you could provide a CLI option to just install the dependencies for a specific config, so I could easily generate my own Docker image without knowing the internals of this project.

Reply:
Oh sorry, I sometimes lose track of who I spoke to about what!

Yeah, the CLI option would be best. I'll update this ticket once I've got around to doing something like that, but I've also got a few other bits in the pipeline that I'm working on for this project, so I'll have to add it to the list.